app/utils/orgs.js

```diff
@@ -68,4 +70,55 @@ const decryptOrgData = (orgKey, data) => {
   return tokenCrypt.decrypt(data, orgKey);
 };
 
-module.exports = { getOrg, verifyAdminOrgKey, encryptOrgData, decryptOrgData};
+const getKmsKeyForOrg = async({ org, metroConf })=>{
+  if(!org?._id){
+    throw new Error('genKmsKeyForOrg requires an org as input');
+  }
+  if(!metroConf){
+    throw new Error('genKmsKeyForOrg requires metroConf in input');
+  }
+  let crn = _.get(org, 'kms.crn');
+  if(crn){
+    // if already has a key, then returns it
+    return crn;
+  }
+  const name = `s3_org_encryption_key_${org._id}`;
+  crn = await genKmsKey({ name, metroConf });
+  const search = {
+    _id: org._id,
+  };
+  const updates = {
+    $set: {
+      'kms.crn': crn,
+    }
+  };
+  await models.Organization.updateOne(search, updates);
+  return crn;
+};
+
+const rotateKmsKeyForOrg = async({ org, metroConf })=>{
+  if(!org || !org._id){
+    throw new Error('pass an org to rotateKmsKeyForOrg()');
+  }
+  if(!metroConf){
+    throw new Error('rotateKmsKeyForOrg requires metroConf in input');
+  }
+  const crn = org.kms.crn;
+  if(!crn){
+    throw new Error('org doesnt have a kms crn');
+  }
+  await rotateKey({ crn, metroConf });
+  const sets = {
+    'kms.lastRotateTime': new Date(),
+  };
+  await models.Organization.updateOne({ _id: org._id }, { $set: sets });
+  return true;
+};
+
+// setTimeout(async()=>{
+//   var org = await models.Organization.findOne({ _id: 'k4TTY77XNPMGjppfW' });
+//   var result = await rotateKmsKeyForOrg({ org });
+//   console.log(77777, org, result)
+// },1)
+
+module.exports = { getOrg, verifyAdminOrgKey, encryptOrgData, decryptOrgData, getKmsKeyForOrg, rotateKmsKeyForOrg };
```

Comment:
```suggestion

module.exports = { getOrg, verifyAdminOrgKey, encryptOrgData, decryptOrgData, getKmsKeyForOrg };
```

Reply:
Suggest removing this unused code.  Instead I suspect it would be better to configure Key Protect to automatically rotate the root key monthly... after consulting with others with more expertise.

Reply:
Lets put it in just incase we need it. If we delete it now, it might get squashed and we'll lose the entire function and have to rewrite it later. idc if you merge and then immediately delete it in a new pr.